src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,49 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.addAnimal(this);
+  }
+
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    delete this.alive[this.alive.indexOf(animal)];
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(herbivore) {
+    if (herbivore.hidden === undefined || herbivore.hidden) {
+      return;
+    }
+
+    herbivore.health -= 50;
```

Comment:
```suggestion
    if (herbivore instanceof Herbivore && !herbivore .hidden) {
     herbivore.health -= 50;
    }
```

We can simplify it a bit